app/tests/spec/models/account.js

```diff
@@ -684,5 +684,38 @@ function (chai, sinon, p, Constants, Assertion, ProfileClient,
           });
       });
     });
+
+    describe('completePasswordReset', function () {
+      it('completes the password reset, signs the user in', function () {
+        account.set('email', EMAIL);
+        account.set('password', PASSWORD);
+        var token = 'token';
+        var code = 'code';
+
+        sinon.stub(fxaClient, 'signIn', function () {
+          return p({});
+        });
+        sinon.stub(fxaClient, 'completePasswordReset', function () {
+          return p(true);
+        });
+        sinon.stub(relier, 'isDirectAccess', function () {
+          return false;
+        });
+
+        return account.completePasswordReset(token, code, relier)
+          .then(function () {
+            assert.isTrue(fxaClient.completePasswordReset.calledWith(
+                EMAIL, PASSWORD, token, code));
+            assert.isTrue(fxaClient.signIn.calledWith(
+                EMAIL,
+                PASSWORD,
+                relier,
+                {
+                  reason: fxaClient.SIGNIN_REASON.PASSWORD_RESET
+                }
+            ));
```

Comment:
We could add an assertion that `account` is updated with session data returned from `fxaClinet.signIn`.